Instance: Add container `finit_module()` syscall interception support (#13151)
canonical/lxd: doc/api-extensions.md

## `vm_disk_io_limits`

Adds the ability to limit disk I/O for virtual machines.

## `ct_linux_kernel_modules_dynamic_load`

> Adds the `linux.kernel_modules.load` container instance configuration option. If `dynamic` value is set then `finit_modules()` syscall will be intercepted and privileged user in the container's user namespace can load Linux kernel modules specified in the
> allow list `linux.kernel_modules`.

Reviewer: ```suggestion
Adds the `linux.kernel_modules.load` container configuration option. If the option is set to `dynamic`, the `finit_modules()` syscall is intercepted and a privileged user in the container's user namespace can load the Linux kernel modules specified in the
allow list `linux.kernel_modules`.
```

Author: Fixed, thanks!

## `vm_disk_io_limits`

Adds the ability to limit disk I/O for virtual machines.

> ## `ct_linux_kernel_modules_dynamic_load`

Reviewer 2: Lets call this `container_syscall_intercept_finit_modules` to align with other intercepted syscalls extensions.

Reviewer 2: Also please can this commit come first in the PR to aid with backporting - allows me to see what the following commits are about.

Reviewer 2: Or should it be `finit_module_parse`?

Reviewer 3: While the `finit_module` syscall designation is technically correct, I had never hear of it before. If we want self descriptive API extension names, maybe one of those would be more friendly to non-kernel devs?

* `container_linux_kernel_modules_load_ondemand`
* `container_linux_kernel_modules_ondemand_loading`

In any case, I think we should have either `dynamic` or `ondemand` (whichever we settle on) in the name to make it easier to link with the actual configuration key/value needed.

Reviewer 2: I think it needs to have interception in it at least so it goes along with the other ones and is clear what is happening.

Author: Have fixed!
